iup/__init__.py

```diff
@@ -831,3 +846,446 @@ def predict(
         self.cumulative_projection = CumulativeUptakeData(self.cumulative_projection)
 
         return self
+
+
+#### prediction output ####
+class Forecast(ValidateData):
+    """
+    Abstract class for all the forecast data type.
+    """
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.validate()
+
+    # Can add validate to enforce the specific schema of Forecast when more Forecast types are added
+    def validate(self):
+        self.assert_type_included(pl.Date)
+        self.assert_type_included(pl.Float64)
+        # has at least 1 column of date and 1 column of estimate
+
+    def assert_type_included(self, datatype):
+        """
+        Verify at least one column has the expected type.
+
+        Parameters
+        columns: List[str]
+            names of columns for which to check type
+        datatype:
+            data type for each listed column
+        """
+        assert any(
+            dtype == datatype for dtype in self.schema.values()
+        ), f"No column is {datatype} type"
+
+
+class QuantileForecast(Forecast):
+    """
+    Class for forecast with quantiles.
+    Save for future.
+    """
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.validate()
+
+    # Must be named as "quantileX.XX" except the date column
+    def validate(self):
+        estimate = self.select(pl.all().exclude(pl.Date))
+        QuantileForecast.assert_column_name_all(estimate, "quantile")
+
+    @staticmethod
+    def assert_column_name_all(estimate, column_name):
+        """
+        Verify that all columns have a pattern with a common name
+
+        Parameters
+        estimate:
+            The data without 'date' type
+        column_name:
+            The common column name
+        """
+        pattern = rf"^{column_name}\d(\.\d+)?$"
+        assert all(
+            [bool(re.match(pattern, col)) for col in estimate.columns]
+        ), f"Not all columns are Column name {column_name}"
```

Comment:
Regular expressions always look complicated to me. I bet there is a simpler-looking way to do this, e.g. 
`assert(all(column_name in col for col in estimate.columns))`

Reply:
This function is used to match the first common word in the column name, it comes in handy in using like line 895. Or we can use [str.find](https://docs.pola.rs/api/python/stable/reference/expressions/api/polars.Expr.str.find.html), which may have better interface.